### PR TITLE
Chapter 7

### PR DESCRIPTION
**Pull request by ShariaHabib**

### CODE REVIEW

File: chapter-7/.idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Editor-based HTTP Client requests
+/httpRequests/
+# Datasource local storage ignored files
+/dataSources/
+/dataSources.local.xml
```
Comment (towfiq-ul) [inline chapter-7/.idea/.gitignore]: you should not push unnecessary code in git. i.e. **.idea**

File: chapter-7/.idea/Chapter-7.iml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (towfiq-ul) [inline chapter-7/.idea/Chapter-7.iml]: you should not push unnecessary code in git. i.e. .idea

File: chapter-7/.idea/misc.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_19" default="true" project-jdk-name="19" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
```
Comment (towfiq-ul) [inline chapter-7/.idea/misc.xml]: you should not push unnecessary code in git. i.e. .idea

File: chapter-7/.idea/modules.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/Chapter-7.iml" filepath="$PROJECT_DIR$/.idea/Chapter-7.iml" />
+    </modules>
+  </component>
+</project>
```
Comment (towfiq-ul) [inline chapter-7/.idea/modules.xml]: you should not push unnecessary code in git. i.e. .idea

**Review by towfiq-ul — CHANGES_REQUESTED** [review]
check the comments and resolve the asked changes.

File: chapter-7/.idea/modules.xml (hunk 4 above)
Comment (ShariaHabib) [reply chapter-7/.idea/modules.xml]: Fixed it vai.

**Review by towfiq-ul — APPROVED** [review]
LGTM

**Pull request merged**